File: решения/java/src/Lab14/stage3/Point.java
package Lab14.stage3;

/**
 * @author AKryukov
 * 05.06.2018
 */
public class Point {
    private double x;
    private double y;
    private String color;

    public Point(double x, double y) {
        this.x = x;
        this.y = y;
        this.color = "white";
    }

    public Point(double x, double y, String color) {
        this.x = x;
        this.y = y;
        this.color = color;
    }

    public void setColor(String value){
        this.color = value;
    }

    public String getColor() {
        return color;
    }

    public static double distanceBetween(double ax, double ay, double bx, double by) {
        double dx = ax - bx;
        double dy = ay - by;
        return Math.sqrt(dx * dx + dy * dy);
    }

    public static double distanceBetween(Point a, Point b) {
        if (a == null) {
            throw new IllegalArgumentException("Начальная точка не инициализирована");
        }
        if (b == null) {
            throw new IllegalArgumentException("Конечная точка не инициализирована");
        }
        return distanceBetween(a.x, a.y, b.x, b.y);
    }

    public double distanceTo(int x, int y) {
        return distanceBetween(this.x, this.y, x, y);
    }

    public double distanceTo(Point p) {
        if (p == null) {
            throw new IllegalArgumentException("Начальная точка не инициализирована");
        }
        return distanceBetween(this, p);
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (other == null) return false;
        if (getClass() != other.getClass()) return false;

        Point point = (Point) other;
        if (x == point.x && y == point.y && color.equals(point.color)) return true;
        return false;
    }

    @Override
    public int hashCode() {
        return (int) (x + 31 * y);
    }

    @Override
    public String toString() {
        return String.format("(%.0f;%.0f;%s)", x, y, color);
    }
}

File: решения/Stage2 cs/CodeProject/Point.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeProject
{
    public class Point
    {
        private double x;
        private double y;
        private String color;

        public Point(double x, double y)
        {
            this.x = x;
            this.y = y;
            this.color = "white";
        }

        public double GetX()
        {
            return x;
        }

        public double GetY()
        {
            return y;
        }

        public static double DistanceBetween(double ax, double ay, double bx, double by)
        {
            double dx = ax - bx;
            double dy = ay - by;
            return Math.Sqrt(dx * dx + dy * dy);
        }

        public static double DistanceBetween(Point a, Point b)
        {
            double dx = a.x - b.x;
            double dy = a.y - b.y;
            return Math.Sqrt(dx * dx + dy * dy);
        }

        public double DistanceTo(double bx, double by)
        {
            return DistanceBetween(this.x, this.y, bx, by);
        }

        public double DistanceTo(Point dest)
        {
            return DistanceBetween(this.x, this.y, dest.x, dest.y);
        }

        public void SetCoordinates(double x, double y)
        {
            this.x = x;
            this.y = y;
        }

        public void SetColor(String value)
        {
            this.color = value;
        }

        public override bool Equals(object other)
        {
            if (other == null) return false;
            if (this == other) return true;
            if (GetType() != other.GetType()) return false;
            Point point = (Point)other;
            if (x == point.x && y == point.y && color.Equals(point.color)) return true;
            return false;
        }

        public override string ToString()
        {
            return String.Format("({0};{1};{2})", x, y, color);
        }
    }
}